src/request_handler.cpp

```diff
@@ -169,70 +316,82 @@ void RequestHandler::on_result_response(ResponseMessage* response) {
   }
 }
 
-void RequestHandler::on_error_response(ResponseMessage* response) {
+void SpeculativeExecution::on_error_response(ResponseMessage* response) {
   ErrorResponse* error =
       static_cast<ErrorResponse*>(response->response_body().get());
 
+  RetryPolicy::RetryDecision decision = RetryPolicy::RetryDecision::return_error();
 
-  switch(error->code()) {
-    case CQL_ERROR_UNPREPARED:
-      on_error_unprepared(error);
-      break;
 
+  switch(error->code()) {
     case CQL_ERROR_READ_TIMEOUT:
-      handle_retry_decision(response,
-                            retry_policy_->on_read_timeout(error->consistency(),
-                                                           error->received(),
-                                                           error->required(),
-                                                           error->data_present() > 0,
-                                                           num_retries_));
+      decision =  request_handler_->retry_policy()->on_read_timeout(request(),
```

Comment:
"Read_timeout: Timeout exception during a read request."  

https://github.com/apache/cassandra/blob/trunk/doc/native_protocol_v3.spec#L982

It explicitly says "write request" and "read request". I'm pretty sure it doesn't have to do we a failed acknowledgement from a replica write as that would manifest as a `Write_timeout`.